src/javascripts/test/unit/Crud/field/maReferenceFieldSpec.js

```diff
@@ -0,0 +1,123 @@
+describe('ReferenceField', function() {
+    var directive = require('../../../../ng-admin/Crud/field/maReferenceField');
+    var ReferenceField = require('admin-config/lib/Field/ReferenceField');
+
+    var $compile, $timeout, scope;
+    const directiveUsage = '<ma-reference-field entry="entry" field="field" value="value"></ma-reference-field>';
+
+    angular.module('myTestingApp', ['ui.select']).directive('maReferenceField', directive);
+
+    beforeEach(function() {
+        angular.mock.module(function($provide) {
+            $provide.service('ReadQueries', function($q) {
+                this.getAllReferencedData = jasmine.createSpy('getAllReferencedData').and.callFake(function() {
+                    var deferred = $q.defer();
+                    deferred.resolve({
+                        'post_id': [
+                            { id: 1, name: 'foo', count: 82 },
+                            { id: 2, name: 'bar', count: 43 },
+                            { id: 3, name: 'qux', count: 71 }
+                        ]
+                    });
+
+                    return deferred.promise;
+                });
+
+                this.getOne = jasmine.createSpy('getOne').and.callFake(function() {
+                    var deferred = $q.defer();
+                    deferred.resolve({ name: 'bar' });
+
+                    return deferred.promise;
+                });
+            });
+        });
+    });
+
+    beforeEach(angular.mock.module('myTestingApp'));
+
+    var MockedReadQueries;
+    beforeEach(inject(function (_$compile_, _$rootScope_, _$timeout_, ReadQueries) {
+        $compile = _$compile_;
+        $timeout = _$timeout_;
+        scope = _$rootScope_;
+        MockedReadQueries = ReadQueries;
+    }));
+
+    beforeEach(function() {
+        scope.field = new ReferenceField('post_id')
+            .targetField({
+                name: () => 'name'
+            })
+            .targetEntity({
+                identifier: () => {
+                    return {
+                        name: () => 'id'
+                    };
+                }
+            });
+    });
+
+    it('should be an ui-select field', function() {
+        var element = $compile(directiveUsage)(scope);
+        scope.$digest();
+
+        var uiSelect = element[0].querySelector('.ui-select-container');
+        expect(uiSelect).toBeTruthy();
+    });
+
+    it('should call remote API when inputting first characters', function () {
```

Comment:
Something bothers me: users can't opt out for ajax autocompletion. That means that if their API doesn't implement the search query param, reference fields cease to work. This is a serious BC break. 

Autocomplete should be opt-in ; the default behavior should still be to fetch all possible values, and only autocomplete locally.